src/bot_methods.js

```diff
@@ -410,29 +296,44 @@ function handleCommandHelp(message) {
   message.channel.send(help);
 }
 
+/**
+ * Handles the 'hitme' command
+ * continuously provides words to translate
+ * @param {Message} message
+ */
 function handleChallengeMode(message) {
-  let newWord = challengeMode(message);
-  console.log(newWord[0]);
-  console.log(newWord[1]);
-  const filter = response => response.content.toLowerCase() == newWord[1].toLowerCase() || response.content.toLowerCase() == 'stop' || response.content.toLowerCase() == 'next';
-  message.channel.send("Translate: " + newWord[0]).then(() => {
-    message.channel.awaitMessages(filter, { maxMatches: 1, time: 15000, errors: ['time'] })
-      .then(collected => {
-        console.log(collected.first().content);
-        if (collected.first().content.toLowerCase() == 'stop') {
+  let wordCheck = false;
+  let newWord = null;
+  while (!wordCheck){ // because some words lack english translations, it loops until a word with a translation is found
+    let randWord = getRandomWord(message); // placeholder
+    newWord = [randWord["word"], randWord["english"]]; // a word and translation
+    if(newWord[1] != null && newWord[1] !== "") {
+      wordCheck = true;
+    }
+  }
+  console.log(newWord[0] + ": " + newWord[1]);
+  const filter = response => response.content.toLowerCase() == newWord[1].toLowerCase() // the filter for accepting user input.
+    || response.content.toLowerCase() == STOP  // if the user types stop, next or the correct word,
+    || response.content.toLowerCase() == NEXT; // it will return true. otherwise false
+  message.channel.send("Translate: **" + newWord[0] + "**. (type " + STOP + "to end)").then(() => {
+    message.channel.awaitMessages(filter, { maxMatches: 1, time: 15000, errors: ['time'] }) // new message collector which accepts one match, in a certain time limit ('time')

```

Comment:
I was thinking, maybe the challenge should stop entirely after some time? in case the user forgot to stop it.
what do you think?

Reply:
good idea